compiler/next/include/chpl/queries/UniqueString-detail.h

```diff
@@ -162,38 +168,24 @@ struct InlinedString {
 // This class is POD and has only the trivial constructor to help the parser
 // (which uses it in a union).
 // All UniqueStrings are actually POD; the difference is that this one
-// does not have a default constructor.
+// does not have a user-defined constructor.
```

Comment:
That comment about unions is not true anymore in C++11 and later.

The restrictions on unions was greatly reduced in C++11:

https://cpp11standard.blogspot.com/2012/11/c11-standard-explained-1-unrestricted.html

https://en.cppreference.com/w/cpp/language/union

A default constructor can be "default declared" by the compiler, or by the user with `Class() = default;`

https://en.cppreference.com/w/cpp/language/default_constructor

When you say it cannot have a "default constructor", it sounds like you're saying that it cannot be default-constructed, such as in:

`ClassType x;`

But that's obviously not true for POD types.

Reply:
The comments in the POD unique strings seem to imply that there was a problem using it with the Parser, and so a separate struct was needed.

This may still be true if C interoperability is required, such as if you are still using C for the Parser, but it's not true with C++11.

Reply:
Right, I can't define my own default constructor for it. I think we are getting hung up on the terminology (does it "have a default constructor" if the compiler generated one but I didn't provide one?). Anyway I got errors adding e.g. my own `PODUniqueString::PodUniqueString()` due to the use in the parser's union. The parser is C++.

Reply:
I think there might be a solution possible, if the parser is C++.

Reply:
I think the most important  distinction is whether you have a _user-defined_ default constructor or not. The compiler-generated one, even if the user _declares_ it `= default`, does not count against you in many situations.

That's why I changed the original language to say _user-defined_. Because you can avoid having a user-defined [default] constructor, but it still gets default-generated by the compiler. So that's not the same as saying that the class has no default constructor (the same concepts apply for the copy constructor, move constructor, copy assignment operator, move assignment operator, destructor).

Reply:
ok